### ISSUE TITLE
singleGAMSfile.R - different behavior of full.gms and code execution in main folder for empty parameters

### ISSUE THREAD

**Issue by flohump** [pik-piam/gms#48]
I recently noticed a problem that only occurs with the full gams file (full.gms) but not if you run the model with gams main.gms from the main folder. The problem is related to empty parameters or tables. We have this in some cases in the code. The problem can be solve by using $OnEmpty and $OffEmpty like here https://github.com/magpiemodel/magpie/blob/develop/modules/57_maccs/on_aug22/input.gms#L23
But before you can solve a problem you need to know that it exists. This is the issue here. In this specific case: If you test non-default settings like c57_macc_version == PBL_2022 locally with gams main.gms it works fine without $OnEmpty and $OffEmpty. You only need these if the singleGAMSfile function is applied. This is not ideal from a user perspective. Is it possible to harmonize the behavior of the two cases?

**Comment by flohump** [comment]
Thanks for linking to the REMIND discussion! This seems to be the right moment for my request ;-)
And if it could solve #49 as well that would be a double dividend.

**Comment by mikapfl** [comment]
Yes, this is one reason why I think the `singleGAMSfile` route is untenable long-term. GAMS has a surprising number of features and users will be confused when they try to use one of the more obscure features and it turns out it is not supported by `singleGAMSfile`. Supporting all features of GAMS compilation properly in `singleGAMSfile` is a lot of work that we are most likely not willing/able to do, so we should instead use `gams action=c` and drop/deprecate `singleGAMSfile`.
